### ISSUE TITLE
El Capitan R 3.4.0 osx binary build broken on CRAN

### ISSUE THREAD

**Issue by edzer** [edzer/sfr#327]
See also https://github.com/ropensci/rerddap/issues/56

Seems that also rgdal binaries are there, but won't work.

**Comment by rsbivand** [comment]
Just notes from looking at the rgdal El Capitan binary package archive (I don't have access to OSX) - the proj-datumgrid files are missing from proj/. The rgdal.so in libs/ is only 127KB, the dSYM only 489KB, they look very small for a static build. As you wrote elsewhere, the Mavericks binary has very much larger files in libs/ (possibly too large if CFLAGS=-O2 and CXXFLAGS=.O2 not imposed to drop the -g flag). The proj-datumgrid files are present. The libs/ sizes for sf Mavericks maybe suggests a similar problem on the GDAL side. For El Capitan, rgeos/libs/ is 2.2MB for the *.so and 14MB for the dSYM, but I guess we know that it was the missing -lgeos and -lm (dropped earlier because geos-config gave the -m breaking value). For Mavericks and rgeos/libs/, *.so is 2.1MB, dSYM is 20MB. So something broke in building the GDAL libraries for El Capitan, maybe, and they just didn't make it through but appear to be there for rgdal, and sf failed because of -lgeos, but would have been stranded on GDAL like rgdal?

**Comment by edzer** [comment]
Yes, that was what I was thinking too.

**Comment by rsbivand** [comment]
Do we have any insight into Simon's build scripts?

**Comment by edzer** [comment]
Build command is

    R CMD INSTALL --dsym --build --configure-args='--with-data-copy=yes --with-proj-data=/usr/local/share/proj' pkg

Working El Capitan binaries for OSX appeared again on CRAN, however both lack the conus etc. files; the Maverick binaries (at least for sf) contain conus.

**Comment by rsbivand** [comment]
Work-around is to copy the Mavericks proj/ folder over the one for El Capitan.

**Comment by s-u** [comment]
@rsbivand the scripts are in https://svn.r-project.org/R-dev-web/trunk/QA/Simon/packages in particular `mk.chk` is what does build+check of a package. But as discussed in the e-mail thread the issue here was that GDAL's build process doesn't work properly for static build on macOS in latest versions so it required manual fix-ups.

**Comment by rsbivand** [comment]
Thanks very much, useful for future reference. Would you like me to post on the gdal-dev list to see if they might document this finding - they are at RC-1 for GDAL 2.2.0 now?

**Comment by s-u** [comment]
I was sending the details to Edzer and I was using GDAL 2.1.3 - I'll send all the details to you so you can raise that with the GDAL team.

**Comment by s-u** [comment]
I can confirm the the 2.2.0-RC1 has fixed the problem, so let's wait for the 2.2.0 release.

**Comment by edzer** [comment]
Thanks a lot for trying this out, Simon!

**Comment by rsbivand** [comment]
I've posted on [this](https://lists.osgeo.org/pipermail/gdal-dev/2017-May/046550.html) thread, so that what Simon established is on the record for GDAL developers. Maybe this is line 54 in 2.2.0/NEWS:

>  * Sort files in static library to make the build reproducible [#6520](https://trac.osgeo.org/gdal/attachment/ticket/6520/sort-files-in-static-library.patch)

but there is so much going on that it is hard to tell. We'll see whether they comment, or care about static builds.

**Comment by edzer** [comment]
For the record, Simon compiles gdal on max osx with
```r
configure --with-pic --disable-shared --enable-static --with-static-proj4=/usr/local/lib/libproj.a --with-liblzma 
```

**Comment by edzer** [comment]
Issue resolved with release [0.4-3](https://cran.r-project.org/web/checks/check_results_sf.html).
